src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -1101,222 +406,211 @@ namespace JSONHelpers
+            TAppZoneHistoryMap appZoneHistoryMap{};
+            auto appLastZones = fancyZonesDataJSON.GetNamedArray(L"app-zone-history");
 
-            json::JsonArray zonesJson = infoJson.GetNamedArray(L"zones");
-            uint32_t size = zonesJson.Size();
-            info.zones.reserve(size);
-            for (uint32_t i = 0; i < size; ++i)
+            for (uint32_t i = 0; i < appLastZones.Size(); ++i)
             {
-                json::JsonObject zoneJson = zonesJson.GetObjectAt(i);
-                const int x = static_cast<int>(zoneJson.GetNamedNumber(L"X"));
-                const int y = static_cast<int>(zoneJson.GetNamedNumber(L"Y"));
-                const int width = static_cast<int>(zoneJson.GetNamedNumber(L"width"));
-                const int height = static_cast<int>(zoneJson.GetNamedNumber(L"height"));
-                CanvasLayoutInfo::Rect zone{ x, y, width, height };
-                info.zones.push_back(zone);
+                json::JsonObject appLastZone = appLastZones.GetObjectAt(i);
+                if (auto appZoneHistory = AppZoneHistoryJSON::FromJson(appLastZone); appZoneHistory.has_value())
+                {
+                    appZoneHistoryMap[appZoneHistory->appPath] = std::move(appZoneHistory->data);
+                }
             }
-            return info;
+
+            return std::move(appZoneHistoryMap);
         }
         catch (const winrt::hresult_error&)
         {
-            return std::nullopt;
+            return {};
         }
     }
 
-    GridLayoutInfo::GridLayoutInfo(const Minimal& info) :
-        m_rows(info.rows),
-        m_columns(info.columns)
+    json::JsonArray SerializeAppZoneHistory(const TAppZoneHistoryMap& appZoneHistoryMap)
     {
-        m_rowsPercents.resize(m_rows, 0);
-        m_columnsPercents.resize(m_columns, 0);
-        m_cellChildMap.resize(m_rows, {});
-        for (auto& cellRow : m_cellChildMap)
+        json::JsonArray appHistoryArray;
+
+        for (const auto& [appPath, appZoneHistoryData] : appZoneHistoryMap)
         {
-            cellRow.resize(m_columns, 0);
+            appHistoryArray.Append(AppZoneHistoryJSON::ToJson(AppZoneHistoryJSON{ appPath, appZoneHistoryData }));
         }
+
+        return appHistoryArray;
     }
 
-    GridLayoutInfo::GridLayoutInfo(const Full& info) :
-        m_rows(info.rows),
-        m_columns(info.columns),
-        m_rowsPercents(info.rowsPercents),
-        m_columnsPercents(info.columnsPercents),
-        m_cellChildMap(info.cellChildMap)
+    TDeviceInfoMap ParseDeviceInfos(const json::JsonObject& fancyZonesDataJSON)
     {
-        m_rowsPercents.resize(m_rows, 0);
-        m_columnsPercents.resize(m_columns, 0);
-        m_cellChildMap.resize(m_rows, {});
-        for (auto& cellRow : m_cellChildMap)
+        try
+        {
+            TDeviceInfoMap deviceInfoMap{};
+            auto devices = fancyZonesDataJSON.GetNamedArray(L"devices");
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.